src/picongpu/include/plugins/PhaseSpace/PhaseSpaceFunctors.hpp

```diff
@@ -104,8 +104,8 @@ namespace picongpu
             int p_bin = int( rel_bin * float_X(num_pbins) );
 
             /* out-of-range bins back to min/max */
-            p_bin >= 0 ? /* do not change p_bin */ : p_bin=0;
-            p_bin < num_pbins ? /* do not change p_bin */ : p_bin=num_pbins-1;
+            p_bin >= 0 ? p_bin /* do not change p_bin */ : p_bin=0;
+            p_bin < num_pbins ? p_bin/* do not change p_bin */ : p_bin=num_pbins-1;
 
```

Comment:
> Maybe the compile can also optimize simple if conditions, we should check it.

We checked this in the past (~CUDA 6) and nvcc is indeed creating way more efficient SIMD ptx for those kinds of short hand assign-ifs.

Generally speaking, `if`s are way more general and harder to optimize then short-hand assigns  (*pattern matching* is *never* your friend. *Looking at you, #pragma based acceleration language extensions!*). But maybe the compiler became more "clever" (has better pattern matching) since then.